spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -272,7 +272,7 @@ private void restart() {
 	protected void doStop() {
 		try {
 			this.active = false;
-			this.lifecycleCondition.await(Math.min(this.stopTimeout, this.receiveTimeout), TimeUnit.MICROSECONDS);
+			this.lifecycleCondition.await(Math.min(this.stopTimeout, this.receiveTimeout), TimeUnit.MILLISECONDS);
```

Comment:
OK. I see now your concern, which we talked about in the #1172. 
So, I'm reworking it to the `min` back and changing test logic do not deal with the same `key` to avoid conflict, even if we don't wait for the proper `stop()` like it is with the `max`.